Narrow scope of clangtidy lintrunner on CI to match lintrunner configs

TovlyFB opened the pull request:
Summary: `--all-files` for the CLANGTIDY lint is too broad, leading it to produce errors on files that should not be linted like `.cuh` files (see [discussion in PR 148936](https://github.com/pytorch/pytorch/pull/148936)). This PR narrows the scope to respect the include and exclude patterns in the `.lintrunner.toml` config.

Test Plan:
1. Apply these changes to D70539649 and [export it to a PR](https://github.com/pytorch/pytorch/pull/148936)
2. Observe that the PR doesn't have any linter errors ([other errors on there are already being looked at](https://fb.workplace.com/groups/pytorch.edge.users/permalink/1721728545364098/) and are separate)

Differential Revision: D71335488

malfet reviewed (changes requested):
Can you provide a test plan of sorts? I.e. introduce a lint violation in one of the files and make sure it detects the failure.

TovlyFB committed:
Narrow scope of clangtidy lintrunner on CI to match lintrunner configs (#149345)

Summary:

`--all-files` for the CLANGTIDY lint is too broad, leading it to produce errors on files that should not be linted like `.cuh` files (see [discussion in PR 148936](https://github.com/pytorch/pytorch/pull/148936)). This PR narrows the scope to respect the include and exclude patterns in the `.lintrunner.toml` config.

Test Plan:
1. Apply these changes to D70539649 and [export it to a PR](https://github.com/pytorch/pytorch/pull/148936)
2. Observe that the PR doesn't have any linter errors ([other errors on there are already being looked at](https://fb.workplace.com/groups/pytorch.edge.users/permalink/1721728545364098/) and are separate)

Differential Revision: D71335488

TovlyFB commented:
> Can you provide a test plan of sorts? I.e. introduce a lint violation in one of the files and make sure it detects the failure.

Test Plan:

1. Remove the line `NOLINTNEXTLINE` on line 24 of `fbcode/caffe2/aten/src/ATen/CPUGeneratorImpl.cpp` in this PR
2. Verify lintrunner fails with clangtidy
3. Revert that `NOLINTNEXTLINE` in this PR
